### PR TITLE
Update `umi-transfer` recipe

### PR DESCRIPTION
**Pull request by MatthiasZepper**
Update `umi-transfer` version from 1.0 to 1.5. Also drop some probably unnecessary build requirements and were once copied from another Rust recipe without much consideration what is really needed.

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

### General instructions

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

### Instructions for avoiding API, ABI, and CLI breakage issues
Conda is able to record and lock (a.k.a. pin) dependency versions used at build time of other recipes.
This way, one can avoid that expectations of a downstream recipe with regards to API, ABI, or CLI are violated by later changes in the recipe.
If not already present in the meta.yaml, make sure to specify `run_exports` (see [here](https://bioconda.github.io/contributor/linting.html#missing-run-exports) for the rationale and comprehensive explanation).
Add a `run_exports` section like this:

```yaml
build:
  run_exports:
    - ...

```

with `...` being one of:

| Case                             | run_exports statement                                               |
| -------------------------------- | ------------------------------------------------------------------- |
| semantic versioning              | `{{ pin_subpackage("myrecipe", max_pin="x") }}`     |
| semantic versioning (0.x.x)      | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}`   |
| known breakage in minor versions | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| known breakage in patch versions | `{{ pin_subpackage("myrecipe", max_pin="x.x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| calendar versioning              | `{{ pin_subpackage("myrecipe", max_pin=None) }}`    |

while replacing `"myrecipe"` with either `name` if a `name|lower` variable is defined in your recipe or with the lowercase name of the package in quotes.

### Bot commands for PR management

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

Note that the <code>@BiocondaBot please merge</code> command is now depreciated. Please just squash and merge instead.

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

File: recipes/umi-transfer/meta.yaml [hunk 1]
```diff
@@ -11,20 +11,16 @@ source:
   
 build:
   number: 0
+  skip: true
```
Comment (martin-g) [inline recipes/umi-transfer/meta.yaml:14]: This disables the build of the recipe.

File: recipes/umi-transfer/meta.yaml [hunk 2]
```diff
@@ -11,20 +11,16 @@ source:
...
-    - {{ compiler('cxx') }}
-    - pkg-config
-    - zlib
-  host:
-    - zlib
-  run:
-
+    - {{ compiler("rust") }}
+    - rust
+    - cargo
```
Comment (martin-g) [inline recipes/umi-transfer/meta.yaml:22]: No need to list `rust` and `cargo`. They come with `{{ compiler("rust") }}`

File: recipes/umi-transfer/meta.yaml (hunk 1 above)
Comment (martin-g) [inline recipes/umi-transfer/meta.yaml:14]:
```suggestion
```

File: recipes/umi-transfer/meta.yaml [hunk 3]
```diff
@@ -11,20 +11,16 @@ source:
...
-    - autoconf
-    - {{ compiler('cxx') }}
-    - pkg-config
-    - zlib
-  host:
-    - zlib
-  run:
-
+    - {{ compiler("rust") }}
+    - rust
```
Comment (martin-g) [inline recipes/umi-transfer/meta.yaml:21]:
```suggestion
```

File: recipes/umi-transfer/meta.yaml (hunk 2 above)
Comment (martin-g) [inline recipes/umi-transfer/meta.yaml:22]:
```suggestion
```

File: recipes/umi-transfer/meta.yaml (hunk 1 above)
Comment (MatthiasZepper) [reply recipes/umi-transfer/meta.yaml]: Oops. It shows that I have no bloody clue of `conda`, and that I have just been copy and pasting stuff from other Rust recipes around.

File: recipes/umi-transfer/meta.yaml (hunk 2 above)
Comment (MatthiasZepper) [reply recipes/umi-transfer/meta.yaml]: Thanks for the valuable tip! I had been looking [for some guidance on Rust tools here](https://bioconda.github.io/contributor/guidelines.html), but apparently it is still too irrelevant for an own section.

**Comment by martin-g** [comment]
```
Downloading https://github.com/SciLifeLab/umi-transfer/archive/refs/tags/v1.5.0.tar.gz[0m
12:48:12 [32mBIOCONDA INFO[0m (OUT) WARNING: Error: HTTP 404 NOT FOUND for url <https://github.com/SciLifeLab/umi-transfer/archive/refs/tags/v1.5.0.tar.gz>[0m
```

The valid url is `https://github.com/SciLifeLab/umi-transfer/archive/refs/tags/v.1.5.0.tar.gz`. Note the `.` between `v` and `1`

File: recipes/umi-transfer/build.sh [hunk 4]
```diff
@@ -1,10 +1,15 @@
...
 # We set CARGO_HOME because we don't pass on HOME to conda-build, thus rendering the default "${HOME}/.cargo" defunct.
 export CARGO_NET_GIT_FETCH_WITH_CLI=true CARGO_HOME="$(pwd)/.cargo"
 
 # build statically linked binary with Rust
-RUST_BACKTRACE=1 cargo install --verbose --root $PREFIX --path .
\ No newline at end of file
+RUST_BACKTRACE=1 cargo install --verbose --locked --root $PREFIX --path .
+
+if command -v ldd &> /dev/null; then
+    echo $(ldd $PREFIX/bin/umi-transfer)
```
Comment (martin-g) [inline recipes/umi-transfer/build.sh:14]: This debug statement could be removed

**Review by martin-g — APPROVED** [review]

**Comment by martin-g** [comment]
@BiocondaBot please fetch artifacts

**Comment by martin-g** [comment]
@BiocondaBot please fetch artifacts

**Comment by martin-g** [comment]
```
mgrigorov in 🌐 euler-arm-22 in /tmp/umi-transfer via 🅒 test123 took 3s 
❯ tar xvf umi-transfer-1.5.0-hba841c1_0.tar.bz2                                                                                                                                                            (test123) 
info/test/run_test.sh
info/run_exports.json
info/files
info/hash_input.json
info/index.json
info/paths.json
info/recipe/build.sh
info/recipe/meta.yaml.template
info/licenses/LICENSE
info/recipe/meta.yaml
info/recipe/conda_build_config.yaml
info/about.json
info/git
bin/umi-transfer
.crates.toml
.crates2.json

mgrigorov in 🌐 euler-arm-22 in /tmp/umi-transfer via 🅒 test123 
❯ file bin/*                                                                                                                                                                                               (test123) 
bin/umi-transfer: ELF 64-bit LSB pie executable, ARM aarch64, version 1 (SYSV), dynamically linked, interpreter /lib/ld-linux-aarch64.so.1, for GNU/Linux 3.7.0, not stripped

mgrigorov in 🌐 euler-arm-22 in /tmp/umi-transfer via 🅒 test123 
❯ ./bin/umi-transfer                                                                                                                                                                                       (test123) 


░░░░░░░░░░░░░░░░░░░░░░░░░░░ SciLifeLab - National Genomics Infrastructure ░░░░░░░░░░░░░░░░░░░░░░░░░░░                                              

https://www.scilifelab.se
https://ngisweden.scilifelab.se
https://github.com/SciLifeLab/umi-transfer                                      

A tool for transferring Unique Molecular Identifiers (UMIs).

Usage: umi-transfer <COMMAND>

Commands:
  external  Integrate UMIs from a separate FastQ file
  help      Print this message or the help of the given subcommand(s)

Options:
  -h, --help     Print help
  -V, --version  Print version
```
LGTM!

File: recipes/umi-transfer/build.sh (hunk 4 above)
Comment (MatthiasZepper) [reply recipes/umi-transfer/build.sh]: I wanted to check if the specified dependencies are truly needed or not. It seems that zlib and the C++ compiler might not be needed.

**Comment by martin-g** [comment]
@MatthiasZepper All looks good to me! Please change the stats to "Ready for review" when you are done!

**Pull request merged**